torch/csrc/jit/passes/reconstruct_scopes.cpp

```diff
@@ -1,83 +1,127 @@
-  void visitNode(Node* n);
+  void visitBlock(Block* b, const std::string& root_scope_string);
+  void visitNode(Node* n, const std::string& root_scope_string);
 
+  std::string getSubModuleName(const Module& module, const std::string& prefix);
   void constructFunctionToModuleMap(const Module& module);
   void constructRelativeNamesForModules(
       const Module& module,
       const std::string& prefix);
 
-  std::string getScopeString(Function* f) const;
+  std::string getScopeString(const InlinedCallStackEntry& frame) const;
 };
 
 void ReconstructScopesPass::constructFunctionToModuleMap(const Module& module) {
   for (auto& method : module.get_methods()) {
-    func_to_module[&method.function()] = module._ivalue();
+    Function* func_ptr = &method.function();
+    if (!has_duplicated_function &&
+        func_to_module.find(func_ptr) != func_to_module.end()) {
+      has_duplicated_function = true;
+    }
+    func_to_module[func_ptr] = module._ivalue();
   }
   for (Module m : module.children()) {
     constructFunctionToModuleMap(m);
   }
 }
 
+std::string ReconstructScopesPass::getSubModuleName(
+    const Module& module,
+    const std::string& prefix) {
+  std::string moduleType = module.type()->str();
+  size_t lastDotIndex = moduleType.rfind('.');
+  if (lastDotIndex != std::string::npos) {
+    moduleType = moduleType.substr(lastDotIndex + 1);
+  }
+  return prefix + "(" + moduleType + ")";
+}
+
 void ReconstructScopesPass::constructRelativeNamesForModules(
     const Module& module,
     const std::string& prefix) {
-  module_names[module._ivalue()] = prefix;
+  module_names[module._ivalue()] = getSubModuleName(module, prefix);
   for (NameModule s : module.named_children()) {
-    constructRelativeNamesForModules(s.value, prefix + "." + s.name);
+    std::string newPrefix = (!has_duplicated_function) ?
+        module_names[module._ivalue()] + "." + s.name : module_names[module._ivalue()] + ".";
+    constructRelativeNamesForModules(s.value, newPrefix);
   }
 }
 
-std::string ReconstructScopesPass::getScopeString(Function* f) const {
+std::string ReconstructScopesPass::getScopeString(const InlinedCallStackEntry& frame) const {
+  Function* f = frame.first;
   if (!func_to_module.count(f)) {
     return "<null (no func in the map)>";
   }
   auto m = func_to_module.at(f);
   if (!module_names.count(m)) {
     return "<null (no module in the map)>";
   }
-  return module_names.at(m) + "." + f->name();
+  std::string scopeString = module_names.at(m) + "." + f->name();
+
+  if (has_duplicated_function) {
+    SourceRange r = frame.second;
+    if (r.source()) {
+      if (auto orig = r.source()->findSourceRangeThatGenerated(r)) {
+        r = *orig;
+      }
+    }
+    if (auto file_line_col = r.file_line_col()) {
+      std::string filename;
+      size_t line, col;
+      std::tie(filename, line, col) = *file_line_col;
+      scopeString += "(" + filename + ":" + c10::to_string(line) + ":" + c10::to_string(col) + ")";
+    }
+  }
```

Comment:
Not clear to me why we should do this? If we decide that this might help, then it is probably best to split it out into a helper function with comments and fn name that suggests what it will do.
But let's first clarify what does this do.

Reply:
That was for adding source range info in the case of a scripted module, as @ZolotukhinM suggested before. But I think I'm gonna remove it now as we decided not to do things differently for the two types of modules.

Reply:
Make it a separate function in that case, atleast, till we resolve it.